### PR TITLE
CONFIG/CUDA: show error messages if nvml not detected

### PR DESCRIPTION
**Pull request by Akshay-Venkatesh**
## What/Why

Configure step to show error if nvml isn't detected on the system.

### CODE REVIEW

**Comment by Akshay-Venkatesh** [comment]
cc @yosefe

**Comment by Akshay-Venkatesh** [comment]
@yosefe builds are failing where nvml is not found now:
```
checking nvml.h usability... no
checking nvml.h presence... no
checking for nvml.h... no
configure: error: nvml header not found. Install appropriate cuda-nvml-devel-X-Y package
```
https://dev.azure.com/ucfconsort/ucx/_build/results?buildId=31149&view=logs&j=e646a2ae-9139-5fa9-0995-85819d80e271&t=40ba7047-ddc0-522c-15d2-d7fcc7934f9b&l=536

Should we just warn instead?

File: config/m4/cuda.m4 [hunk 1]
```diff
@@ -51,10 +51,24 @@ AS_IF([test "x$cuda_checked" != "xyes"],
...
+         AS_IF([test "x$cuda_happy" = "xyes"],
+               [],
+               [AC_MSG_ERROR([nvml header not found. Install appropriate cuda-nvml-devel-X-Y package])])
+
+         # Check nvml library
          AS_IF([test "x$cuda_happy" = "xyes"],
                [AC_CHECK_LIB([nvidia-ml], [nvmlInit],
                              [CUDA_LIBS="$CUDA_LIBS -lnvidia-ml"], [cuda_happy="no"])])
 
+         AS_IF([test "x$cuda_happy" = "xyes"],
```
Comment (yosefe) [inline config/m4/cuda.m4:68]: 1. Pls do it like this:
 https://github.com/openucx/ucx/blob/29aa1f2abdb39a6d32f8b00fd2a65e2321139e70/src/ucs/configure.m4#L121
2. err message is:
`libnvidia-ml.so is not found. Please install appropriate nvidia driver package`
3. can add it as the "else" part of the AC_CHECK_LIB in line 65 (`test x$with_cuda != -xguess`)

**Comment by yosefe** [comment]
ok to test

**Review by yosefe — APPROVED** [review]

**Comment by Akshay-Venkatesh** [comment]
Tests are failing with:
```
/usr/bin/ld: cannot find -levent_core
/usr/bin/ld: cannot find -levent_pthreads
collect2: error: ld returned 1 exit status
make[1]: *** [shmem_pingpong] Error 1
make[1]: Leaving directory `/scrap/azure/agent-01/AZP_WORKSPACE/2/s/build-test/test/mpi'
make: *** [install-recursive] Error 1
##[error]Bash exited with code '2'.
```

**Comment by yosefe** [comment]
@Akshay-Venkatesh pls squash without rebase

**Review by yosefe — APPROVED** [review]

**Pull request merged**